playground/frontend/CONTRIBUTE.md

```diff
@@ -0,0 +1,135 @@
+│   │   └── embedded_playground     # embedded version of the editor
+├──  test               # folder with unit tests
+├── pubspec.yaml        # infromation about application and dependencies
+├── build.gradle        # gradle tasks for playground frontends
+├── gradle.properties   # default properties for project
+```
+
+## State Management
+
+Playground uses [provider](https://pub.dev/packages/provider) package for state management. We have
+top level providers like `ThemeProvider`, common page level provider `PlaygroundPageProviders` which
+contains decoupled playground page state and module providers like `OutputPlacementState`.
+
+For quick start up, please take a look
+to [this guide](https://docs.flutter.dev/development/data-and-backend/state-mgmt/simple)
+
+## Configuration
+
+The app could be configured using gradle task (e.g. api url)
+
+```
+cd beam
+./gradlew :playground:frontend:createConfig
+```
+
+The command generates [./lib/config.g.dart](./lib/config.g.dart) file with constants from
+gradle.properties which can be used in the app
+
+To add a new constant do the following steps:
+
+- Add a new line with const to the `:playground:frontend:createConfig` task
+  inside [build.gradle](./build.gradle);
+- Update [gradle.properties](./gradle.properties);
+- Run task to test it and commit [./lib/config.g.dart](./lib/config.g.dart) file.
+
+## Theming
+
+Playground app supports light and dart themes. Component themes are declared
+in [theme.dart](./lib/config/theme.dart) file.
+
+To use specific color inside component you can use helper `ThemeColors` utility:
+
+`ThemeColors.of(context).greyColor`
+
+[colors.dart](./lib/constants/colors.dart) contains color declarations.
+
+## Adding a new page
+
+To add a new page do the following steps:
+
+- Add a page component to the `pages` folder
+- Add a url to the [routes.dart](./lib/pages/routes.dart) class as a static property
+- Add a case clause to the same class with your component
+
+```
+case Routes.page_url:
+    return Routes.renderRoute(const PageComponent());
+```
+
+## Accessibility
+
+Please, read the following guides about the accessibility:
+
+- [Flutter Doc](https://docs.flutter.dev/development/accessibility-and-localization/accessibility)
+- [Medium Article](https://medium.com/flutter-community/a-deep-dive-into-flutters-accessibility-widgets-eb0ef9455bc)
+
+## Unit testing
+
+Unit tests are on `tests` folder. It keeps the project structure from the `lib` folder
+with `_test.dart` file postfix.
+
+We are using standard flutter library for unit testing
+and [Mockito](https://pub.dev/packages/mockito) for generating mocks. To generate mocks for class
+you need to add `@GenerateMocks([ExampleClient])` annotation to a test file and
+run `$ flutter pub run build_runner build` command.
+
+Playground tests may be run using next commands:
+
+`$ flutter test`
+
+## Generated Files
+
+All generated files (generated api clients, mocks) should be published to the repository.
```

Comment:
should we document how to generate files? Or is this documented elsewhere? Should we link to that place then?

Reply:
This is a general section for generated files. All commands are listed in other sections (e.g. Unit testing has command to generate mocks). I'm not sure that links will be helpful here.